pkg/nvml/lib.go

```diff
@@ -255,3 +268,36 @@ func (l *library) updateVersionedSymbols() {
 		nvmlVgpuInstanceGetLicenseInfo = nvmlVgpuInstanceGetLicenseInfo_v2
 	}
 }
+
+// LibraryOption represents a functional option to configure the underlying NVML library
+type LibraryOption func(*library)
+
+// WithLibraryPath provides an option to set the library name to be used by the NVML library.
+func WithLibraryPath(path string) LibraryOption {
+	return func(l *library) {
+		l.path = path
+	}
+}
+
+// SetLibraryOptions applies the specified options to the NVML library.
+// If this is called when a library is already loaded, and error is raised.
+func SetLibraryOptions(opts ...LibraryOption) error {
+	libnvml.Lock()
+	defer libnvml.Unlock()
+	if libnvml.dl != nil {
+		return errLibraryAlreadyLoaded
+	}
+
+	for _, opt := range opts {
+		opt(&libnvml)
+	}
+
+	if libnvml.path == "" {
+		libnvml.path = defaultNvmlLibraryName
+	}
+	if libnvml.flags == 0 {
```

Comment:
How do you unset all flags?

Reply:
Is that a use case that is expected? If so we should use a `*int` to detect whether it is set or not.

Note it's not currently possible to set the flags through the options, so we could address this in a follow-up.

Reply:
Note sure. I doubt anyone will ever override this (beyond adding to it) so its probably fine.

Reply:
Let's reevaluate if it ever becomes relevant.